src/BarracksServer/Network/PacketHandler.cs

```diff
@@ -692,7 +705,25 @@ public void CB_PET_COMMAND(IBarracksConnection conn, Packet packet)
 			var characterId = packet.GetLong();
 			var command = packet.GetByte(); // 0 : revive request; 1 : delete pet request.
 
-			// ...
+			var companion = conn.Account.GetCompanionById(companionId);
+			if (companion == null)
+			{
+				Log.Warning("CB_PET_COMMAND: Companion not found by id '{0}' received from '{1}'.", companionId, conn.Account.Name);
+				return;
+			}
+
+			companion.CharacterDbId = characterId;
+			switch (command)
+			{
+				// Delete
+				case 1:
```

Comment:
When turning types into actionable values, it's generally better to create an enum for better readability and type safety. It would also be useful to log unknown values in the default case or as result of an `Enum.IsDefined` check, so we know when new, previously unknown values, come in.